src/apm.coffee

```diff
@@ -49,6 +50,15 @@ module.exports =
         unless fs.existsSync(appLocation)
           appLocation = '/usr/share/atom/resources/app.asar'
         process.nextTick -> callback(appLocation)
+      else # Windows
+        pattern = "/Users/#{process.env.USERNAME}/AppData/Local/atom/app-+([0-9]).+([0-9]).+([0-9])/resources/app.asar"
+        glob pattern, null, (er, appLocations) ->
```

Comment:
```suggestion
          pattern = "/Users/#{process.env.USERNAME}/AppData/Local/atom/app-+([0-9]).+([0-9]).+([0-9])/resources/app.asar"
          appLocations = glob.sync(pattern, null)
```

Reply:
Isn't async supposed to be better for performance, or something? That's what I keep hearing. Otherwise I would have reached for `glob.sync`.

Reply:
I would not be surprised though if it was a second or less difference though, if any.

Reply:
This function is not async (getResourcePath), and so it expects the output right away. Making this async may result in unexpected behavior. If this function was async we could use `glob`.

General note: Async code is only useful when the code is reading/writing/fetching/running something from outside of the current process. Nodejs is single-threaded and so making a CPU intensive work code async does not help at all.

Reply:
I checked the usages of this function, nowhere the return is used. However, CoffeeScript 1 does not support async functions, and so there is no performance benefits between the two:

```
❯ benchmark { .\bin\apm.cmd --version }

Title/no.    : 1
Average (ms) : 935.28272
Count        : 10
Sum (ms)     : 9352.8272
Maximum (ms) : 976.9314
Minimum (ms) : 919.1504

```

```
❯ benchmark { .\bin\apm.cmd --version }

Title/no.    : 1
Average (ms) : 936.52319
Count        : 10
Sum (ms)     : 9365.2319
Maximum (ms) : 1017.2834
Minimum (ms) : 909.3007
```